acme_diags/driver/qbo_driver.py

```diff
@@ -0,0 +1,178 @@
+
+    # Amplitude
+    AMP0 = 2 * abs(X0[0:int(NFFT0 / 2 + 1)])
+    # Power spectral density
+    PSDx0 = AMP0 ** 2 / L0
+    Tend0 = T0 * L0
+    # Total spectral power
+    Pxf0 = Tend0 * np.sum(PSDx0)
+    # TODO: Because why?
+    # Need to flip original period, AMP, and PSDx0 because
+    period0_flipped = period0[::-1]
+    AMP0_flipped = AMP0[::-1]
+    PSDx0_flipped = PSDx0[::-1]
+
+    AMPnew0 = np.interp(periodNew, period0_flipped[:-1], AMP0_flipped[:-1])
+    PSDxnew0 = np.interp(periodNew, period0_flipped[:-1], PSDx0_flipped[:-1])
+    return PSDxnew0, AMPnew0
+
+
+def run_diag(parameter):
+    variables = parameter.variables
+    # The region will always be 5S5N
+    region = '5S5N'
+    test_data = utils.dataset.Dataset(parameter, test=True)
+    ref_data = utils.dataset.Dataset(parameter, ref=True)
+    for variable in variables:
+        if parameter.print_statements:
+            print('Variable={}'.format(variable))
+        test_var = test_data.get_timeseries_variable(variable)
+        ref_var = ref_data.get_timeseries_variable(variable)
+        qbo_region = default_regions.regions_specs[region]['domain']
+
+        test_region = test_var(qbo_region)
+        ref_region = ref_var(qbo_region)
+
+        test = {}
+        ref = {}
+
+        # Diagnostic 1: average over longitude & latitude to produce time,height array of u field:
+        # https://en.wikipedia.org/wiki/Quasi-biennial_oscillation
+        # U = "monthly-mean, zonal-mean equatorial zonal wind"
+        test['qbo'], test['level'] = process_u_for_time_height(test_region)
+        ref['qbo'], ref['level'] = process_u_for_time_height(ref_region)
+
+        # Diagnostic 2: calculate and plot the amplitude of wind variations with a 20-40 month period
+        # TODO: What is pspec?
+        test['pspec'], test['amp'] = get_20to40month_fft_amplitude(np.squeeze(np.array(test['qbo'])), test['level'])
+        ref['pspec'], ref['amp'] = get_20to40month_fft_amplitude(np.squeeze(np.array(ref['qbo'])), ref['level'])
+
+        # Diagnostic 3: calculate the Power Spectral Density
+        # Pre-process data to average over lat,lon,height
+        x_test = process_u_for_power_spectral_density(test_region)
+        x_ref = process_u_for_power_spectral_density(ref_region)
+        # Calculate the PSD and interpolate to period_new. Specify periods to plot
+        period_new = np.concatenate((np.arange(2, 33), np.arange(34, 100, 2)), axis=0)
+        test['PSDx_new'], test['amp_new'] = get_psd_from_deseason(x_test, period_new)
+        ref['PSDx_new'], ref['amp_new'] = get_psd_from_deseason(x_ref, period_new)
+
+        parameter.var_id = variable
+        parameter.main_title = 'QBO index, amplitude, and power spectral density for {}'.format(variable)
+        parameter.viewer_descr[variable] = parameter.main_title
+
+        test['name'] = parameter.test_name
+        ref['name'] = parameter.ref_name
+
+        # TODO: Check the below works properly by using ncdump on Cori
```

Comment:
Confirm these are saved properly.

Reply:
@chengzhuzhang Only `level` and `qbo` are actually stored as nc files, with the other fields being saved in json. Using `ncdump` on Cori,  the `level` nc files appear to be empty `{}` while the `qbo` nc files appear to hold data.